pages/k8s/charm-kubernetes-master.md

```diff
@@ -0,0 +1,629 @@
+---
+wrapper_template: "kubernetes/docs/base_docs.html"
+markdown_includes:
+  nav: "kubernetes/docs/shared/_side-navigation.md"
+context:
+  title: "kubernetes-master charm"
+  description: Kubernetes-master Charm reference
+keywords: kubernetes-master, charm, config
+tags: [reference]
+sidebar: k8smain-sidebar
+permalink: charm-kubernetes-master.html
+layout: [base, ubuntu-com]
+toc: False
+---
+[Kubernetes](http://kubernetes.io/) is an open source system for managing
+application containers across a cluster of hosts. The Kubernetes project was
+started by Google in 2014, combining the experience of running production
+workloads combined with best practices from the community.
+
+The Kubernetes project defines some new terms that may be unfamiliar to users
+or operators. For more information please refer to the concept guide in the
+[getting started guide](https://kubernetes.io/docs/home/).
+
+This charm is an encapsulation of the Kubernetes master processes and the
+operations to run on any cloud for the entire lifecycle of the cluster.
+
+This charm is built from other charm layers using the Juju reactive framework.
+The other layers focus on specific subset of operations making this layer
+specific to operations of Kubernetes master processes.
+
+# Deployment
+
+This charm is not fully functional when deployed by itself. It requires other
+charms to model a complete Kubernetes cluster. A Kubernetes cluster needs a
+distributed key value store such as [Etcd](https://coreos.com/etcd/) and the
+kubernetes-worker charm which delivers the Kubernetes node services. A cluster
+requires a Software Defined Network (SDN), a Container Runtime such as
+[containerd](https://jaas.ai/u/containers/containerd), and Transport Layer
+Security (TLS) so the components in a cluster communicate securely.
+
+Please take a look at the [Canonical Distribution of Kubernetes](https://jujucharms.com/canonical-kubernetes/)
```

Comment:
`s/CDK/Charmed Kubernetes` and update link to https://jaas.ai/charmed-kubernetes

Reply:
I think this got skipped